### ISSUE TITLE
Problems during training

### ISSUE THREAD

**Issue by VViiepan** [AmadeusITGroup/Moire-Pattern-Detection#25]
Hello! Thanks for sharing！I encountered some problems during fitting CNN model。
Hope for your help！
The following is the training log：
  1/102 [..............................] - ETA: 19:24 - loss: 0.6917 - accuracy: 0.5938
  2/102 [..............................] - ETA: 8:29 - loss: 0.6912 - accuracy: 0.5938 
  3/102 [..............................] - ETA: 8:18 - loss: 0.6808 - accuracy: 0.6042
  4/102 [>.............................] - ETA: 8:10 - loss: 0.6791 - accuracy: 0.5859
  5/102 [>.............................] - ETA: 8:07 - loss: 0.6739 - accuracy: 0.6062
  6/102 [>.............................] - ETA: 8:00 - loss: 0.6623 - accuracy: 0.6146
  7/102 [=>............................] - ETA: 7:56 - loss: 0.6399 - accuracy: 0.6339
  8/102 [=>............................] - ETA: 7:51 - loss: 0.6412 - accuracy: 0.6328
  9/102 [=>............................] - ETA: 7:46 - loss: 0.6255 - accuracy: 0.6493
 10/102 [=>............................] - ETA: 7:40 - loss: 0.6102 - accuracy: 0.6656
 11/102 [==>...........................] - ETA: 7:36 - loss: 0.6136 - accuracy: 0.6619
 12/102 [==>...........................] - ETA: 7:30 - loss: 0.6283 - accuracy: 0.6589
 13/102 [==>...........................] - ETA: 7:25 - loss: 0.6245 - accuracy: 0.6635
 14/102 [===>..........................] - ETA: 7:20 - loss: 0.6191 - accuracy: 0.6652
 15/102 [===>..........................] - ETA: 7:15 - loss: 0.6211 - accuracy: 0.6708
 16/102 [===>..........................] - ETA: 7:09 - loss: 0.6234 - accuracy: 0.6699
 17/102 [====>.........................] - ETA: 7:03 - loss: 0.6240 - accuracy: 0.6728
 18/102 [====>.........................] - ETA: 6:58 - loss: 0.6233 - accuracy: 0.6753
 19/102 [====>.........................] - ETA: 6:53 - loss: 0.6240 - accuracy: 0.6694
 20/102 [====>.........................] - ETA: 6:47 - loss: 0.6217 - accuracy: 0.6656
 21/102 [=====>........................] - ETA: 6:42 - loss: 0.6227 - accuracy: 0.6652
 22/102 [=====>........................] - ETA: 6:37 - loss: 0.6223 - accuracy: 0.6605
 23/102 [=====>........................] - ETA: 6:32 - loss: 0.6200 - accuracy: 0.6603
 24/102 [======>.......................] - ETA: 6:27 - loss: 0.6188 - accuracy: 0.6576
 25/102 [======>.......................] - ETA: 6:22 - loss: nan - accuracy: 0.6637   
 26/102 [======>.......................] - ETA: 6:16 - loss: nan - accuracy: 0.6623
 27/102 [======>.......................] - ETA: 6:11 - loss: nan - accuracy: 0.6562
 28/102 [=======>......................] - ETA: 6:06 - loss: nan - accuracy: 0.6607
 29/102 [=======>......................] - ETA: 6:00 - loss: nan - accuracy: 0.6595
 30/102 [=======>......................] - ETA: 5:55 - loss: nan - accuracy: 0.6583
 31/102 [========>.....................] - ETA: 5:50 - loss: nan - accuracy: 0.6562
 32/102 [========>.....................] - ETA: 5:45 - loss: nan - accuracy: 0.6562
 33/102 [========>.....................] - ETA: 5:39 - loss: nan - accuracy: 0.6544
 34/102 [=========>....................] - ETA: 5:34 - loss: nan - accuracy: 0.6535
 35/102 [=========>....................] - ETA: 5:29 - loss: nan - accuracy: 0.6509
 36/102 [=========>....................] - ETA: 5:24 - loss: nan - accuracy: 0.6432
 37/102 [=========>....................] - ETA: 5:19 - loss: nan - accuracy: 0.6385
 38/102 [==========>...................] - ETA: 5:14 - loss: nan - accuracy: 0.6398
39/102 [==========>...................] - ETA: 5:09 - loss: nan - accuracy: 0.6410
 40/102 [==========>...................] - ETA: 5:04 - loss: nan - accuracy: 0.6398
 41/102 [===========>..................] - ETA: 4:59 - loss: nan - accuracy: 0.6418
 42/102 [===========>..................] - ETA: 4:54 - loss: nan - accuracy: 0.6384
 43/102 [===========>..................] - ETA: 4:49 - loss: nan - accuracy: 0.6344
 44/102 [===========>..................] - ETA: 4:44 - loss: nan - accuracy: 0.6385
 45/102 [============>.................] - ETA: 4:39 - loss: nan - accuracy: 0.6396
 46/102 [============>.................] - ETA: 4:34 - loss: nan - accuracy: 0.6372
 47/102 [============>.................] - ETA: 4:29 - loss: nan - accuracy: 0.6343
******
 98/102 [===========================>..] - ETA: 19s - loss: nan - accuracy: 0.6279
 99/102 [============================>.] - ETA: 14s - loss: nan - accuracy: 0.6294
100/102 [============================>.] - ETA: 9s - loss: nan - accuracy: 0.6309 
101/102 [============================>.] - ETA: 4s - loss: nan - accuracy: 0.6315
102/102 [==============================] - ETA: 0s - loss: nan - accuracy: 0.6315
Epoch 1: val_loss did not improve from inf

102/102 [==============================] - 519s 5s/step - loss: nan - accuracy: 0.6315 - val_loss: nan - val_accuracy: 0.5861
Why does the loss became nan after the 24th iteration, and what means 'val_loss did not improve from inf' ?
Did I do something wrong during training or data preparation?
Hope for your help！

**Comment by VViiepan** [comment]
Problems have been solved.
